px-dropdown.html

```diff
@@ -129,478 +127,537 @@
     },
     /**
-     * This function appropriately positions the dropdown within the bounds given.
+     * The sort function used by the dom-repeat inside of iron-selector to
+     * sort the items by either 'key' or 'val' based on `sortMode`.
      */
-    positionWithinBounds: function (parentBoundingRect) {
-      var content = Polymer.dom(this).querySelector('px-dropdown-content'),
-        dropdown = content.$.dropdown,
-        dropcell = this.contentAnchor ? this.contentAnchor : this.$.dropcell,
-        dropcellRect = dropcell.getClientRects()[0],
-        dropdownRect,
-        dropdownBottomPoint,
-        sizeAbove,
-        sizeBelow;
-
-      //reset content size and get some values
-      content.resetHeight();
-      dropdownRect = dropdown.getBoundingClientRect();
-      dropdownBottomPoint = dropdownRect.bottom,
-        sizeAbove = dropdownRect.top - parentBoundingRect.top - parseInt(dropcellRect.height),
-        sizeBelow = parentBoundingRect.bottom - dropdownRect.top;
-
-      //if we can't fit it below
-      if (dropdownBottomPoint > parentBoundingRect.bottom) {
+    _computeSort: function (a, b) {
+      var sortValue = 0;
+      if (!this.sortMode) {
+        return -1;
+      }
 
-        //can we fit it above ?
-        if (sizeAbove > dropdownRect.height) {
-          content.adjustHeight();
-          this._setTopPosition();
-        } else {
+      if (this.sortMode && sortValue === 0) {
+        if (this.sortMode === 'key') {
+          sortValue = a.key - b.key;
+        }
+        if (this.sortMode === 'val') {
+          var nameA = a.val.toUpperCase(),
+            nameB = b.val.toUpperCase();
 
-          //fit it where we have the most space
-          if (sizeAbove > sizeBelow) {
-            content.sizeHeight(sizeAbove - 1);
-            this._setTopPosition();
-          } else {
-            content.sizeHeight(sizeBelow - 1);
+          if (nameA < nameB) {
+            sortValue--;
+          }
+          if (nameA > nameB) {
+            sortValue++;
           }
         }
       }
+
+      return sortValue;
     },
     /**
-     * This function changes the position of the content area to be above
-     * the dropcell, instead of the default below.
+     * Adds or removes the focused class for keyboard navigation.
      */
-    _setTopPosition: function () {
-      var content = Polymer.dom(this).querySelector('px-dropdown-content'),
-        dropdown = content.$.dropdown,
-        dropdownRect = dropdown.getClientRects()[0],
-        dropcell = this.contentAnchor ? this.contentAnchor : this.$.dropcell,
-        dropcellRect = dropcell.getClientRects()[0],
-        newTop = (parseInt(dropdown.offsetTop) - parseInt(dropdownRect.height) - parseInt(dropcellRect.height)) + 'px';
-
-      dropdown.style.top = newTop;
-      this.above = true;
+    _setFocusedOption: function(newOption, oldOption) {
+      this._focusedOption = newOption;
+      if(newOption) {
+        newOption.classList.add('focused');
```

Comment:
Prefer the use of `this.toggleClass()` rather than manually manipulating the classList .